src/test/java/com/google/javascript/cl2dts/lends.js

```diff
@@ -0,0 +1,19 @@
+goog.provide('lends');
+goog.provide('lends.A');
+
+/**
+ * @constructor
+ */
+lends.A = function() {
+  /** @type {string} */
+  this.a = "";
+};
+
+goog.object.extend(lends.A, /** @lends {lends.A} */ {b: 1});
+
+// lends manages to avoid the deduping of fields, and as far as closure
+// is concerned there is a field a on the prototype and on the instance.
+// Also the presense of @lends extends the type signature of A.prototype.
+goog.object.extend(lends.prototype.A, /** @lends {lends.A.prototype} */ {a: 1, c: true});
```

Comment:
that is correct `lends.A.prototype` is the correct usecase.

You confusion is very warranted. In fact half of the documentation for @lends is spent trying to clarify that point. https://developers.google.com/closure/compiler/docs/js-for-compiler?hl=en

Basically, @lends works beyond the type system, and it works against the `raw` JS objects and the keys we expect on them. In that case `lends.A` would tell closure that `a` is a static method. So working as expected, though expected is quite odd.

Overall @lends is quite a confusing feature, and really only makes sense when you try to type `goog.object.extend` method of dynamically building constructors.